contracts/root/depositManager/DepositManager.sol

```diff
@@ -33,6 +35,15 @@ contract DepositManager is DepositManagerStorage, IDepositManager, IERC721Receiv
     depositEther();
   }
 
+  // next 2 functions are just to update cache (since childChain and stateSender are frequently used variables)
+  function changeChildChain() public {
```

Comment:
it should be push-based because we might not remember to pull it and screw things up.